File: datasets/IR-Plag/original-java-rest-kotlin/case-07/plagiarized/L4/09/Level4.kt
package IR

import java.util.*

/*
* To change this license header, choose License Headers in Project Properties.
* To change this template file, choose Tools | Templates
* and open the template in the editor.
*/
/**
 *
 * @author FD8DBE9073AFCC0504CD6901E1091CAD
 */
object Level4 {
    @JvmStatic
    fun main(args: Array<String>) {
        val `in` = Scanner(System.`in`)
        val matriks = Array(4) { DoubleArray(4) }
        var sum = 0.0

        print("Enter a 4 by 4 matrix row by row: ")
        for (b in 0..3) {
            for (k in 0..3) {
                matriks[b][k] = `in`.nextDouble()
            }
        }

        for (x in matriks.indices) {
            sum += matriks[x][x]
        }

        print("Sum of the elements in the major diagonal is $sum")
    }
}

File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/plagiarized/L4/09/Level4.py
import numpy as np

matrix = []
print("Enter a 4 by 4 matrix row by row: ", end="")
for _ in range(4):
    row = [float(x) for x in input().split()]
    matrix.append(row)

matrix = np.array(matrix)
diagonal_sum = np.trace(matrix)

print(f"Sum of the elements in the major diagonal is {diagonal_sum}")

